exchanges/kline/kline.go

```diff
@@ -349,8 +350,29 @@ func (k *Item) ConvertToNewInterval(newInterval Interval) (*Item, error) {
 	if len(candles) == 0 {
 		return nil, fmt.Errorf("%w to %v no candle data", ErrInsufficientCandleData, newInterval)
 	}
+
 	var target int
 	for x := range k.Candles {
+		if k.Candles[x].Open == 0 &&
+			k.Candles[x].High == 0 &&
+			k.Candles[x].Low == 0 &&
+			k.Candles[x].Close == 0 &&
+			k.Candles[x].Volume == 0 {
+			// This candle is padding, there is nothing to apply to the new
+			// interval candle. We can skip this candle and move on to the next.
+			if (x+1)%oldIntervalsPerNewCandle == 0 {
+				// Note: Below checks the length of the proceeding slice so we can
+				// break instantly if we cannot make an entire candle. e.g. 60 min
+				// candles in an hour candle and we have 59 minute candles left.
+				// This entire procession is cleaved.
+				if len(k.Candles[x:])-1 < oldIntervalsPerNewCandle {
+					break
+				}
+				target++
```

Comment:
If possible I'd like to see a test which does hit this, because I'm curious how it interacts with the other `target++` below in case of craziness